src/main/java/com/swirlds/recordserver/TokensService.java

```diff
@@ -239,4 +241,210 @@ private void getTokenById(ServerRequest request, ServerResponse response) {
     }
 
+/**********************
+### listTokenBalancesById specifying only a token id
+GET http://localhost:8080/api/v1/tokens/629591/balances?limit=99
+### listTokenBalancesById specifying a token id and an account id
+GET http://localhost:8080/api/v1/tokens/629591/balances?account.id=644972
+### listTokenBalancesById specifying a token id and a range of account ids
+GET http://localhost:8080/api/v1/tokens/629591/balances?account.id=gt:644000&account.id=lt:644999&limit=1000
+### listTokenBalancesById specifying a token id and a range of account ids, sorted in descending order
+GET http://localhost:8080/api/v1/tokens/629591/balances?account.id=gt:644000&account.id=lt:644999&limit=1000&order=desc
+### listTokenBalancesById specifying a token id and an account id and a particular balance amount
+GET http://localhost:8080/api/v1/tokens/629591/balances?account.id=644972&account.balance=500000000
+### listTokenBalancesById specifying a token id and an account id and a particular timestamp -- not yet working
+GET http://localhost:8080/api/v1/tokens/629591/balances?account.id=644972&timestamp=1644887406787336960
+ **********************/
+    private void listTokenBalancesById(ServerRequest request, ServerResponse response) {
+        // this routine has much in common with BalanceService.getDefaultMessageHandler(); they should be kept in sync.
+
+        final Optional<String> accountIdQueryParam = request.queryParams().first("account.id");
+        final Optional<String> accountBalanceQueryParam = request.queryParams().first("account.balance");
+        final Optional<String> orderParam = request.queryParams().first("order");
+        final Optional<String> publicKeyQueryParam = request.queryParams().first("account.publickey");
+        final Optional<String> timestampQueryParam = request.queryParams().first("timestamp"); // TODO (MYK): can be a list
+        final Optional<String> limitParam = request.queryParams().first("limit");
+
+        // TODO if publicKeyParam is set then need to query accounts table to get account ID from public key, then look
+
+        // build and execute query
+        final List<QueryParamUtil.WhereClause> whereClauses = new ArrayList<>();
+        final int limit = parseLimitQueryString(limitParam);
+        System.out.println("listTokenBalancesById: limit = " + limit);
+        final QueryParamUtil.WhereClause accountWhereClause = (accountIdQueryParam.isPresent() ?
+                QueryParamUtil.parseQueryString(QueryParamUtil.Type._long, "account_id", accountIdQueryParam.get()) :
+                null);
+        System.out.println("accountWhereClause = " + accountWhereClause);
+        // restrict query to only the tokenId passed in as a path parameter
+        final String tokenId = request.path().segments().get(0);
+        accountBalanceQueryParam.ifPresent(s -> whereClauses.add(
+                QueryParamUtil.parseQueryString(QueryParamUtil.Type._long, "balance", s)));
```

Comment:
It's not just equals for balance, there can be other operators too. You might want to handle the other  >,< cases too.

Reply:
That line of code handles less than, less than or equals, greater than, and greater than or equals.  Not 100% sure about "not equals", but I know it handles the rest.